### DISCUSSION TITLE
Error: P1001: Can't reach database server

### DISCUSSION THREAD

**Discussion by x945** [vercel/next.js#68146] [Help]
### Summary

I'm following the tutorial on Vercel guides to build an app with Next.js, Prisma, and Vercel Postgre. After I have created a new database on Vercel and connected the project to it, I try run the command `npx prisma db push` but I get the output `Error: P1001: Can't reach database server`

prisma.schema
```
// schema.prisma

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url = env("POSTGRES_PRISMA_URL") // uses connection pooling
  directUrl = env("POSTGRES_URL_NON_POOLING") // uses a direct connection
}

model Post {
  id        String     @default(cuid()) @id
  title     String
  content   String?
  published Boolean @default(false)
  author    User?   @relation(fields: [authorId], references: [id])
  authorId  String?
}

model User {
  id            String       @default(cuid()) @id
  name          String?
  email         String?   @unique
  createdAt     DateTime  @default(now()) @map(name: "created_at")
  updatedAt     DateTime  @updatedAt @map(name: "updated_at")
  posts         Post[]
  @@map(name: "users")
}
```

The output when running the command `npx prisma db push`
```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "verceldb", schema "public" at "ep-little-salad-a1yltpjv.ap-southeast-1.aws.neon.tech"

Error: P1001: Can't reach database server at `ep-little-salad-a1yltpjv.ap-southeast-1.aws.neon.tech:5432`

Please make sure your database server is running at `ep-little-salad-a1yltpjv.ap-southeast-1.aws.neon.tech:5432`.
FAIL
```

### Additional information

[How to Build a Fullstack App with Next.js, Prisma, and Vercel Postgres](https://vercel.com/guides/nextjs-prisma-postgres)

**Comment by Juneezee** [comment]
Are you sure you created your Vercel Postgres database correctly? Also make sure you copied `POSTGRES_PRISMA_URL` and `POSTGRES_URL_NON_POOLING` correctly from **Settings -> Environment Variables**. See https://vercel.com/docs/storage/vercel-postgres/quickstart#review-what-was-created for more info.

**Reply by x945** [reply]
Yes, the Vercel Postgres database was set up correctly following the steps in the tutorial. I have used vercel env pull .env to pull the environment variables.

**Reply by kamranwds** [reply]
I have the same error with prisma after following the tutorial, is there any solution?

**Reply by anshul-031** [reply]
me too is facing same issue, any solution yet?

**Reply by marcosdipaolo** [reply]
+1
